src/test/functional/Superescalar/iteratelist.spec.ts

```diff
@@ -0,0 +1,72 @@
+import { expect, beforeEach, test } from 'vitest'
+import { Code } from '../../../core/Common/Code';
+import { Superescalar } from '../../../core/Superescalar/Superescalar';
+import { SuperescalarStatus } from '../../../core/Superescalar/SuperescalarEnums';
+import { FunctionalUnitType } from '../../../core/Common/FunctionalUnit';
+import { codeInput, memContent } from "../code/recorrelista";
+
+
```

Comment:
I'll write this here since it's the first test file that shows in the review.

Some concerns about the implementation of tests:
1. It might be preferable to wrap the unit tests into `describe` blocks. See more about this approach [here](https://vitest.dev/guide/features#in-source-testing).
2. Each test would then be converted into `it` blocks. Describing tests through this approach makes it so descriptions are conventional, starting with `should ...`. I'll add a comment as an example down below, in one of the test cases.

A final remark about using Vitest is that tests can be in the source code, like Rust does. What do you think about this feature? We might decide to go this way in the future. https://vitest.dev/guide/features#in-source-testing

Reply:
Skipping this as mentioned here https://github.com/etsiiull/SIMDE/pull/78#issuecomment-1831669875